SEP-0012.md

```diff
@@ -0,0 +1,537 @@
+            coordinates are desired. ``axes=None`` implies all axes will be
+            returned.
+
+        edges
+            If `True` then instead of returning the coordinates of the pixel
+            centers the coordinates of the pixel edges will be returned.
+
+        wcs
+            The WCS object to used to calculate the world coordinates.
+            Although technically this can be any valid WCS, it will typically be
+            ``self.wcs``, ``self.extra_coords``, or ``self.combined_wcs``, combing both
+            the WCS and extra coords.
+            Default=self.wcs
+
+        Returns
+        -------
+        axes_coords
+            `~astropy.units.Quantity` or iterable thereof for all requested
+            world axes, units determined by the wcs.  # TODO: Should we mandate namedtuple?
+
+        Example
+        -------
+        >>> NDCube.all_world_coords_values(('lat', 'lon'))
+        >>> NDCube.all_world_coords_values(2)
+
+        """
+
+    @abc.abstractmethod
+    def crop(self,
+             lower_corner: Iterable[Union[Any, None]],
+             upper_corner: Iterable[Union[Any, None]],
+             wcs: Optional[Union[BaseHighLevelWCS, ExtraCoordsABC]] = None
+             ) -> NDCubeABC:
+        """
+        Crops an NDCube given real world coords of lower and upper corners of a region of interest.
+
+        Parameters
+        ----------
+        lower_corner
+            An iterable of length-1 astropy higher level objects, e.g. SkyCoord,
+            representing the real world coordinates of the lower corner of
+            the region of interest.
+            These are input to ``wcs.world_to_array_index``
+            so their number and order must be compatible with the API of that method.
+            Alternatively, `None`, can be provided instead of a higher level object.
+            In this case, the corresponding array axes will be cropped starting from
+            0th array index.
+
+        upper_corner
+            An iterable of length-1 astropy higher level objects, e.g. SkyCoord,
+            representing the real world coordinates of the upper corner of
+            the region of interest.
+            These are input to ``wcs.world_to_array_index``
+            so their number and order must be compatible with the API of that method.
+            Alternatively, `None`, can be provided instead of a higher level object.
+            In this case, the corresponding array axes will be cropped to include
+            the final array index.
+
+        wcs
+            The WCS to use to calculate the pixel coordinates based on the input.
+            Will default to the ``.wcs`` property if not given. While any valid WCS
+            could be used it is expected that either the ``.wcs`` or
+            ``.extra_coords.wcs`` properties will be used.
+
+        Returns
+        -------
+        result: NDCube
+        """
+
+    @abc.abstractmethod
+    def crop_by_values(self,
```

Comment:
Maybe nicer names for this and the above method would be `crop_along_coords` and `crop_along_array` or similar, to distinguish them?

Reply:
Oh hold on, do these methods do exactly the same thing, but with different input? If so, they should be cross-referenced in their docstrings.

Reply:
both the crop methods crop by the world coordinate, they just vary as to if they use the object notation or not. I will clarify the docstring.